File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        totalApples = reduce(lambda acc, cur: acc + cur, apple)
        sortedBoxes = sorted(capacity, reverse = True)
        count = 0

        for box in sortedBoxes:
            totalApples -= box
            count += 1

            if totalApples <= 0:
                break;
        
        return count
      

File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.js
/**
 * @param {number[]} apple
 * @param {number[]} capacity
 * @return {number}
 */
var minimumBoxes = function(apple, capacity) {
  let totalApples = apple.reduce((acc, cur) => acc + cur, 0);
  const sortedBoxes = capacity.sort((a, b) => b - a);
  let count = 0;
  
  for (let i = 0; i < sortedBoxes.length; i++) {
    totalApples -= sortedBoxes[i];
    count++;
    
    if (totalApples <= 0) break;
  }
  
  return count;
};
